lib/alloc_prof.c

```diff
@@ -0,0 +1,177 @@
+/* -*- C -*- */
+/*
+ * Copyright (c) 2015-2022 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+
+
+/**
+ * @addtogroup memory
+ *
+ * @{
+ */
+
+#define M0_TRACE_SUBSYSTEM M0_TRACE_SUBSYS_MEMORY
+#include "lib/trace.h"
+#include "lib/alloc_prof.h"
+#include "lib/mutex.h"
+
+#if USE_ALLOC_PROF
+
+enum { SLOTS = 1000 };
+
+extern char __start_m0allocprof;
+extern char __stop_m0allocprof;
+
+static struct m0_alloc_prof *begin = (void *)&__start_m0allocprof;
+static struct m0_alloc_prof *end = (void *)&__stop_m0allocprof;
+
+static struct m0_alloc_prof_tls ptdata[AP_NR][SLOTS] = {};
```

Comment:
This variable is supposed to be per thread according to the documentation for struct m0_alloc_prof_tls.